ui/lib/pki/addon/components/pki-role-generate.ts

```diff
@@ -33,9 +35,16 @@ export default class PkiRoleGenerate extends Component<Args> {
     evt.preventDefault();
     this.errorBanner = '';
     const { model, onSuccess } = this.args;
+    const { isValid, state, invalidFormMessage } = model.validate();
+
     try {
-      yield model.save();
-      onSuccess();
+      if (isValid) {
+        yield model.save();
+        onSuccess();
+      } else {
+        this.modelValidations = state;
```

Comment:
dust 🧹  - Lines 45 and 46 could be moved above line 41 (but still inside the try..catch) and this else block could be removed. 

Just as an FYI to show other ways we handle validations - A single return line for `if (!isValid) return` could also be used here. Chelsea's work in `pki-sign-intermediate-form.ts` is a great example!